src/cpp/session/modules/SessionRSConnect.cpp

```diff
@@ -459,8 +475,21 @@ Error getRmdPublishDetails(const json::JsonRpcRequest& request,
    if (path.exists() && (path.hasExtensionLowerCase(".rmd") || 
                          path.hasExtensionLowerCase(".md")))
    {
+      // check to see if this is a website directory
       if (r_util::isWebsiteDirectory(path.parent()))
          websiteDir = path.parent().absolutePath();
+      else
+      {
+         // if the folder containing the R Markdown or Markdown file wasn't itself
+         // a website folder, then see if this is a project, and the project-root 
+         // is a website; this supports source files stored in subfolders of a 
+         // website project
+         projects::ProjectContext& context = session::projects::projectContext();
```

Comment:
For website projects there is the concept of the "build dir" for the website (as opposed to the project directory). So you might need to check `context.config().websitePath` (that might or might not be the right reference but it's something like that)